app/pages/notifications/notification-section.jsx

```diff
@@ -143,13 +144,13 @@ export default class NotificationSection extends Component {
     const { notificationsCounter } = this.context;
     const { user } = this.props;
     const { notifications } = this.state;
-    const relatedNotifications = notifications.filter(
-      notification => notification.source_id === readNotification.source_id
-    );
-    relatedNotifications.push(readNotification);
-    const readRequests = relatedNotifications
+    const relatedNotifications = notifications.filter(notification => (
+      notification.subscription_id === readNotification.subscription_id
```

Comment:
I think this needs to incorporate the `source_id` and `source_type` as well. I'm not sure the `subscription_id` matches across different users. 

I'm not entirely sure which attributes you need to 'relate' the notifications together but my testing resulted in the above attributes. I think you need to handle @ mentions, comments and other forms of notification events.

Reply:
Based on [this gist](https://gist.github.com/mcbouslog/37f9c33b1ba12497e7d0b8a184f7765b) I think comments, replies and mentions in the same discussion should be marked as read if the user goes to said discussion, otherwise I think other notifications shouldn't be grouped (moderation reports, data requests) but marked as read individually. This is just my opinion though, of course open to input.